app/src/components/popups/BusStopPopup.vue

```diff
@@ -117,9 +148,34 @@ watch(
   { immediate: true },
 );
 
+const tripSubscriptions = new Set<string>();
+
+// watch arrivals and subscribe to trips
+watch(
+  stop,
+  async (newStop, oldStop) => {
+    if (newStop === null || newStop.arrivals === null || newStop.arrivals === oldStop?.arrivals) {
+      return;
+    }
+
+    // eslint-disable-next-line no-restricted-syntax
+    for await (const arrival of newStop.arrivals) {
+      if (!tripSubscriptions.has(arrival.tripId)) {
+        tripSubscriptions.add(arrival.tripId);
+        await subscribe(`data.map.trip.${arrival.tripId}`, trips);
```

Comment:
This seems to be adding quite a few subscriptions. In case all busses have left, those subscriptions are still open, right? I am not sure if you really need to await them in here as you don't care about their order. I guess you could just void those calls and speed up the loading process crazily.

Reply:
- [ ] TODO: unsubscribe from trips that already passed the stop

@anbraten Voiding the subscribe call did not really speed things up. Maybe we have some locks in the collector preventing subscribing to multiple topics at once?

Reply:
The slowdown is due to how we call the collector when a new subscription is created:

1st subscription: queries 1 trip
2nd subscription: queries 2 trips
...
10th subscription: queries 10 trips

Reply:
Solved by #213